src/test/app/NFToken_test.cpp

```diff
@@ -4166,6 +4168,48 @@ class NFToken_test : public beast::unit_test::suite
             BEAST_EXPECT(env.balance(buyer, gwXAU) == gwXAU(0));
             BEAST_EXPECT(env.balance(broker, gwXAU) == gwXAU(1050));
 
+            // Burn the NFT so the next test starts with a clean state.
+            env(token::burn(buyer, nftID));
+            env.close();
+        }
+        // broker has a balance less than the seller offer
+        {
+            checkOwnerCountIsOne({issuer, minter, buyer, broker}, __LINE__);
```

Comment:
I really like that you added this unit test.  Thank you very much!

That said, I think we need to have the test run both with and without the amendment.  It's not hard to do, but it can be a bit messy.  I have an example for one possible approach to the change in the top-most commit here:

https://github.com/scottschurr/rippled/commits/gregw-brokered-mode-nft-patch

But don't feel that you need to follow that example.  There may be better ways to do it.